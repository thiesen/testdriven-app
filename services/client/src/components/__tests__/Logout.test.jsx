import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import Logout from '../Logout';

const logoutUser = jest.fn();

describe('<Logout/>', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Logout logoutUser={logoutUser}/>);
    const element = wrapper.find('p');

    expect(element.length).toBe(1);
    expect(element.get(0).props.children[0]).toContain('You are now logged out.');
  });

  test('snapshot', () => {
    const tree = renderer.create(
      <Router><Logout logoutUser={logoutUser}/></Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
