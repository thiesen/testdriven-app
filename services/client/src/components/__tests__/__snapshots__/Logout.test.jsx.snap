// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout/> snapshot 1`] = `
<div>
  <p>
    You are now logged out. Click 
    <a
      href="/login"
      onClick={[Function]}
    >
      here
    </a>
     to log back in.
  </p>
</div>
`;
